### PR TITLE
retire DeveloperMenuState

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Retire `DeveloperMenuState` and read the config with `Bundle`.

### CODE REVIEW

**Comment by typfel** [comment]
The usage  of `enableDeveloperMenu` has lost its meaning since it's now used control all sorts of debug related controls. 

I would propose to rename to to `developerModeEnabled`.

File: WireCommonComponents/AppCenter.swift [hunk 1]
```diff
@@ -57,6 +57,9 @@ public extension Bundle {
         return Bundle.appMainBundle.infoForKey("UseAppCenter") == "1"
     }
 
+    static var enableDeveloperMenu: Bool {
```
Comment (typfel) [inline WireCommonComponents/AppCenter.swift]: I think `AppCenter.swift` is not the right place for this property.

Comment (billypchan) [reply WireCommonComponents/AppCenter.swift]: moved to a new file under Wire iOS target

File: Wire-iOS/Sources/Components/Settings.swift [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
-#else
-    return NO;
-#endif
+extension Settings {
+    @objc
+    func startLogging() {
+        #if targetEnvironment(simulator)
+        loadEnabledLogs()
+        #else
+        ZMSLog.startRecording(withIsInternal: Bundle.enableDeveloperMenu)
```
Comment (typfel) [inline Wire-iOS/Sources/Components/Settings.swift]:
```suggestion
        ZMSLog.startRecording(withIsInternal: Bundle.developerModeEnabled)
```

?

**Pull request merged**